server/api/api.go

```diff
@@ -2166,6 +2170,216 @@ func (a *API) handleUploadFile(w http.ResponseWriter, r *http.Request) {
 	auditRec.AddMeta("fileID", fileID)
 	auditRec.Success()
 }
+func (a *API) handleTeamBoardsInsights(w http.ResponseWriter, r *http.Request) {
+	// swagger:operation GET /teams/{teamID}/boards/insights handleTeamBoardsInsights
+	//
+	// Returns team boards insights
+	//
+	// ---
+	// produces:
+	// - application/json
+	// parameters:
+	// - name: teamID
+	//   in: path
+	//   description: Team ID
+	//   required: true
+	//   type: string
+	// - name: time_range
+	//   in: query
+	//   description: duration of data to calculate insights for
+	//   required: true
+	//   type: string
+	// - name: page
+	//   in: query
+	//   description: page offset for top boards
+	//   required: true
+	//   type: string
+	// - name: per_page
+	//   in: query
+	//   description: limit for boards in a page.
+	//   required: true
+	//   type: string
+	// security:
+	// - BearerAuth: []
+	// responses:
+	//   '200':
+	//     description: success
+	//     schema:
+	//       type: array
+	//       items:
+	//         "$ref": "#/definitions/BoardInsight"
+	//   default:
+	//     description: internal error
+	//     schema:
+	//       "$ref": "#/definitions/ErrorResponse"
+
+	vars := mux.Vars(r)
+	teamID := vars["teamID"]
+	userID := getUserID(r)
+	query := r.URL.Query()
+	timeRange := query.Get("time_range")
+
+	if !a.permissions.HasPermissionToTeam(userID, teamID, model.PermissionViewTeam) {
```

Comment:
This only makes sense in plugin mode, so I would do a check here to verify that only takes place in plugin mode because if not is always going to fail when it tries to get the user timezone. The same happens in the other API call implemented. Both has to return a Not Implement HTTP status code in desktop or standalone server.

Reply:
https://github.com/mattermost/focalboard/pull/3596/commits/cf42959b53526f54ca58373951dee8488bd80d2a